api/v1alpha1/humiopdfrenderservice_types.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2020 Humio https://humio.com
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// HumioPdfRenderServiceStateUnknown is the unknown state of the PDF rendering service.
+	HumioPdfRenderServiceStateUnknown = "Unknown"
+	// HumioPdfRenderServiceStateExists is the Exists state of the PDF rendering service.
+	HumioPdfRenderServiceStateExists = "Exists"
+	// HumioPdfRenderServiceStateNotFound is the NotFound state of the PDF rendering service.
+	HumioPdfRenderServiceStateNotFound = "NotFound"
+)
+
+// HumioPdfRenderServiceSpec defines the desired state of HumioPdfRenderService
+type HumioPdfRenderServiceSpec struct {
+	// Image is the Docker image to use for the PDF rendering service.
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Image string `json:"image"`
+
+	// Replicas is the number of desired Pod replicas.
+	Replicas int32 `json:"replicas"`
+
+	// Port is the port the service listens on.
+	// +optional
+	Port int32 `json:"port,omitempty"`
+
+	// Resources defines the resource requests and limits for the container.
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// Env allows to specify environment variables for the service.
+	// +optional
+	Env []corev1.EnvVar `json:"env,omitempty"`
```

Comment:
I added a similar comment below as well, but I think we should call this `EnvironmentVariables` to remain consistent with this project.